web/client/selectors/layers.js

```diff
@@ -20,8 +20,10 @@ import { mapSelector } from './map';
 import { getSelectedMapView } from './mapviews';
 import { mergeViewLayers } from '../utils/MapViewsUtils';
 import { currentLocaleSelector } from "../selectors/locale";
+import { hasFields } from '../components/TOC/fragments/LayerFields';
 
 export const layersSelector = ({layers, config} = {}) => layers && isArray(layers) ? layers : layers && layers.flat || config && config.layers || [];
+export const nonBackgroundLayersSelector = (state) => layersSelector(state).filter(l => l.group !== "background").filter(hasFields);
```

Comment:
This utility function is coming from a plugin, you should not import it from it.
Copy the function for the moment. then TOC can reuse the selector if needed, on a future refector.

Your selector is not `nonBackgroundLayersSelector` anymore. The name is wrong at this stage.

You can call it `wfsBackedLayersSelector` in `selectors/layers`
or put it in `selectors/geoProcessingTools.js` and call it `compatibleLayersSelector`

Reply:
I should add it to to LayerUtils (it regards our layers, not the WFS service itself).
Thinking more deeply, maybe is easier name it "hasWFSService". 
Notice that you will need in your selector to select layers that: 
- are not background layers
- has a WFS Service

A little specific maybe, so I suggest to create your own selector in your selector's file. Do not generalize too much until you have enough use case to verify your abstraction is good.